packages/x-date-pickers/src/DesktopDateTimePicker/DesktopDateTimePicker.tsx

```diff
@@ -26,26 +28,52 @@ const DesktopDateTimePicker = React.forwardRef(function DesktopDateTimePicker<TD
   // Props with the default values common to all date time pickers
   const defaultizedProps = useDateTimePickerDefaultizedProps<
     TDate,
+    DateOrTimeViewWithMeridiem,
     DesktopDateTimePickerProps<TDate>
   >(inProps, 'MuiDesktopDateTimePicker');
 
-  const viewRenderers: PickerViewRendererLookup<TDate | null, DateOrTimeView, any, {}> = {
-    day: renderDateViewCalendar,
-    month: renderDateViewCalendar,
-    year: renderDateViewCalendar,
-    hours: null,
-    minutes: null,
-    seconds: null,
-    ...defaultizedProps.viewRenderers,
-  };
+  const timeSteps = { hours: 1, minutes: 5, seconds: 5, ...defaultizedProps.timeSteps };
+  const shouldUseNewRenderer =
+    !defaultizedProps.viewRenderers || Object.keys(defaultizedProps.viewRenderers).length === 0;
+
+  const viewRenderers: PickerViewRendererLookup<TDate | null, DateOrTimeViewWithMeridiem, any, {}> =
+    // we can only ensure the expected two-column layout if none of the renderers are overridden
+    shouldUseNewRenderer
+      ? {
+          day: renderDesktopDateTimeView,
+          month: renderDesktopDateTimeView,
+          year: renderDesktopDateTimeView,
+          hours: renderDesktopDateTimeView,
+          minutes: renderDesktopDateTimeView,
+          seconds: renderDesktopDateTimeView,
+          meridiem: renderDesktopDateTimeView,
+        }
+      : {
+          day: renderDateViewCalendar,
+          month: renderDateViewCalendar,
+          year: renderDateViewCalendar,
+          hours: null,
+          minutes: null,
+          seconds: null,
+          meridiem: null,
+          ...defaultizedProps.viewRenderers,
+        };
   const ampmInClock = defaultizedProps.ampmInClock ?? true;
+  // add "accept" action only when the new date time view renderers are used
+  const actionBarActions: PickersActionBarAction[] = defaultizedProps.viewRenderers
```

Comment:
Don't change the default for anyone with custom/provided renderers.

Reply:
Shoudn't it be `!shouldUseNewRenderer` here ?

Reply:
Yes, it should/could for correctness. 👌